qutebrowser/browser/commands.py

```diff
@@ -1737,6 +1739,12 @@ def fullscreen(self, leave=False):
 
         window = self._tabbed_browser.widget.window()
 
+        if enter:
+            if window.isFullScreen():
+                return
+            window.setWindowState(Qt.WindowFullScreen)
+            return
+
         if not window.isFullScreen():
             window.state_before_fullscreen = window.windowState()
```

Comment:
You should set `window.state_before_fullscreen` (so it's properly restored when exiting fullscreen) as well with `--enter`.

Reply:
Moved the set above the enter code